twitter_class.py

```diff
@@ -0,0 +1,57 @@
+import json
+import base64
+import requests
+from requests_oauthlib import OAuth1
+
+from environment_variables import get_env
+
+
+class Twitter:
+    method = {"twitter_upload_pic": "https://upload.twitter.com/1.1/media/upload.json",
+              "twitter_create_post": "https://api.twitter.com/1.1/statuses/update.json",
+              "twitter_get_users_posts": "https://api.twitter.com/1.1/statuses/user_timeline.json",
+              "delete_tweet_by_id": "https://api.twitter.com/1.1/statuses/destroy/{}.json"}
+
+    methodType = {"twitter_upload_pic": "post",
+                  "twitter_create_post": "post",
+                  "twitter_get_users_posts": "get",
+                  "delete_tweet_by_id": "post"}
+
+    auth = OAuth1(get_env("TWITTER_CONSUMER_KEY"),
+                  client_secret=get_env("TWITTER_CONSUMER_SECRET"),
+                  resource_owner_key=get_env("TWITTER_ACCESS_KEY"),
+                  resource_owner_secret=get_env("TWITTER_ACCESS_SECRET"),signature_type="body")
+
+    def request(self, method_name, **kwargs):
+        params = {}
+        result = None
+        for key in kwargs:
+            params[key] = kwargs[key]
+        if self.methodType[method_name] == "post":
+            if method_name == "delete_tweet_by_id":  # that method has different query structure than others
+                result = requests.post(self.method[method_name].format(kwargs["tweet_id"]), data=params, auth=self.auth)
```

Comment:
You can do data=**kwargs, without intermediary “params”

Reply:
Also, you can do .format on every url - if there are no {} inside, it will have no effect